lib/specValidator.js

```diff
@@ -51,610 +178,508 @@ class SpecValidator {
     return;
   }
 
-  constructErrorObject(code, message, innerErrors) {
+  constructErrorObject(code, message, innerErrors, skipValidityStatusUpdate) {
     let err = {
       code: code,
       message: message,
     }
     if (innerErrors) {
       err.innerErrors = innerErrors;
     }
-    this.updateValidityStatus();
+    if (!skipValidityStatusUpdate) {
+      this.updateValidityStatus();
+    }
     return err;
-}
+  }
 
-  initialize(options, callback) {
+  initialize() {
     let self = this;
-    if (!callback && typeof options === 'function') {
-      callback = options;
-      options = {skipInitialization: false};
-    }
-
-    if (options && !options.skipInitialization) {
-      //process.chdir(path.dirname(self.specPath));
-      RefParser.bundle(self.specPath, function(bundleErr, bundleResult){
-        if (bundleErr) {
-          let msg = `Error occurred in parsing the spec "${self.specPath}". \t${bundleErr.message}.`;
-          bundleErr.code = 'PARSE_SPEC_ERROR';
-          bundleErr.message = msg;
-          self.specValidationResult.resolveSpec = {};
-          self.specValidationResult.resolveSpec.error = bundleErr;
-          console.log(`${bundleErr.code} - ${bundleErr.message}`);
-          self.updateValidityStatus();
-          return callback(bundleErr);
+    return utils.parseJson(self.specPath).then(function (result) {
+      self.specInJson = result;
+      self.unifyXmsPaths();
+      return self;
+    }).then(function () {
+      return self.makeModelsStricter();
+    }).then(function() {
+      return Object.keys(self.specInJson.definitions).map(function(definitionName){
+        if (self.specInJson.definitions[definitionName].allOf) {
+          delete self.specInJson.definitions[definitionName].allOf;
```

Comment:
While resolving allOf, there were cases where 

A -> B -> C
        ^
        |
       D

I found that deleting allOfs was giving inconsistent results. So made it a separate step.